test/test_segment_reductions.py

```diff
@@ -0,0 +1,37 @@
+import torch
+from torch.testing._internal.common_device_type import (
+    instantiate_device_type_tests,
+    onlyCPU,
+    dtypes,
+)
+from torch.testing._internal.common_utils import (
+    TestCase,
+    run_tests,
+)
+
+
+class TestSegmentReductions(TestCase):
+    @onlyCPU
+    @dtypes(torch.half, torch.bfloat16, torch.float, torch.double)
+    def test_max_simple_1d(self, device, dtype):
+        lengths = torch.tensor([1, 2, 3], device=device)
```

Comment:
usually it'll be a good idea to use hypothesis to test more cases w/ random number, and use numpy to write a ref implementation?

Reply:
We typically don't use hypothesis in torch testsuite

Reply:
I see, is it because of pulling extra dependencies? JUst try to think how to make the test stronger and cover corner cases; Maybe we can use torch.randint to randomize the data (btw in the past we do see some corner cases which fails the job, e.g. when input data is empty)

Reply:
We usually specifically test for corner-case inputs instead of hoping that they'd be randomly generated

Reply:
I still have todo to add more testing, documentation and benchmark in follow-up diffs. So this is still WIP.

Facebook:
Once we have high confidence on the op, only then will switch Dper3's LengthsMax and variations to segment_reduce.

Reply:
I still have todo to add more testing, documentation and benchmark in follow-up diffs. So this is still WIP.